### PR TITLE
[edward] Fixed per-channel wsy/wsz loss multipliers

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Wall-shear y/z axes are the dominant binding constraint:
- val wsy=9.100% vs AB-UPT target 3.65% (5.45pp gap)
- val wsz=10.869% vs AB-UPT target 3.63% (7.24pp gap)
- val wsx=7.109% vs target 5.35% (only 1.76pp gap)

Across all Wave 1-3 runs the gap is consistent: wsy/wsz are ~3-4× above target while wsx, cp, vol_p are within 1-2× of target. Prior attempts to attack this via complex auto-balancing (edward PR #84 uncertainty weighting REJECTED, PR #137 GradNorm DIVERGED) failed because the controllers were unstable.

The simplest possible attack: hard-code per-channel loss multipliers `wsy×3.0, wsz×5.0` directly in the per-channel rel-L2 / MSE loss. No controller, no learned weights, no clamping — just a fixed scalar multiplier on the wsy and wsz channels of the wall-shear loss term. This is the PR #236 idea (closed non-response).

If this works, expected ep30 val_abupt ~6.5-7.0% with wsy/wsz dropping by 30-50%.

## Instructions

This requires a small `train.py` / loss-function code change. Find the wall-shear loss term (likely an MSE or rel-L2 over a 3-channel tensor). Add per-channel multipliers:

```python
# Pseudocode for the wall-shear loss site
ws_loss_per_channel = ...  # shape [..., 3] for wsx, wsy, wsz
ws_channel_weights = torch.tensor([1.0, args.wsy_weight, args.wsz_weight], device=ws_loss_per_channel.device)
ws_loss = (ws_loss_per_channel * ws_channel_weights).mean()
```

Add CLI flags `--wsy-weight` (default 1.0) and `--wsz-weight` (default 1.0) so the change is backward-compatible (default behavior unchanged).

Run a 2-trial sweep using `--wandb-group bengio-wave4-wsyz-multipliers`.

**Trial A — wsy×3.0, wsz×5.0 (target-gap-proportional)**:
```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 \
  --no-use-ema --lr 3e-4 --lr-cosine-t-max 30 --no-compile-model \
  --wsy-weight 3.0 --wsz-weight 5.0 \
  --kill-thresholds "35000:val_primary/abupt_axis_mean_rel_l2_pct<20" \
  --wandb-group bengio-wave4-wsyz-multipliers
```

**Trial B — wsy×2.0, wsz×3.0 (conservative)**: same with `--wsy-weight 2.0 --wsz-weight 3.0`.

Trial A first. Launch Trial B once Trial A passes ep5 healthy (val_abupt < 12% at ep5).

**Gates**: ep10<10.0, ep20<8.0, ep30<7.0 for Trial A. Report ep30 val_abupt + per-axis decomposition. The key signal is wsy and wsz absolute values dropping vs the 9.10/10.87 baseline.

## Baseline

Current best (from `target/BASELINE.md`):
- `val_primary/abupt_axis_mean_rel_l2_pct`: **7.2091** (ep30 ckpt, run `m9775k1v`)
- `val_primary/wall_shear_y_rel_l2_pct`: 9.100 (target 3.65 — binding)
- `val_primary/wall_shear_z_rel_l2_pct`: 10.869 (target 3.63 — binding)
- `val_primary/wall_shear_x_rel_l2_pct`: 7.109 (target 5.35)
- `val_primary/surface_pressure_rel_l2_pct`: 4.802
- `val_primary/volume_pressure_rel_l2_pct`: **4.166**

**Caveat**: Watch that wsx and surface_pressure don't regress more than the wsy/wsz gain. If wsx degrades by >1pp, the recipe is net-negative.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR — **URGENT: Launch required now**

This PR has been open for >24 hours with no student launch comment. The per-channel wsy×3.0/wsz×5.0 fixed loss multiplier experiment is a high-priority Wave 4 assignment — directly attacking our binding constraint.

**Immediate action required.** Implement the per-channel multiplier and launch Trial A:

```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 \
  --no-use-ema --lr 3e-4 --lr-cosine-t-max 30 --no-compile-model \
  --wsy-weight 3.0 --wsz-weight 5.0 \
  --kill-thresholds "35000:val_primary/abupt_axis_mean_rel_l2_pct<20" \
  --wandb-group bengio-wave4-wsyz-multipliers
```

Add CLI flags `--wsy-weight` (default 1.0) and `--wsz-weight` (default 1.0) to `train.py` at the per-channel wall-shear loss site. Post the W&B run ID and confirm the implementation in a PR comment.

Gates: ep10<10.0, ep20<8.0, ep30<7.0.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #255 because ADVISOR: Closing after two URGENT notices with no response in >48 hours. Experiment was per-channel wsy×3.0/wsz×5.0 fixed loss multipliers — high-priority binding-constraint attack. Hypothesis remains valid and will be reassigned..
